app/src/main/java/unipd/se18/ocrcamera/ActionBarFragment.java

```diff
@@ -0,0 +1,92 @@
+package unipd.se18.ocrcamera;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentActivity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+
+/**
+ * this fragment want to manage the action bar in the activity
+ * in the xml files you need:
+ * action_bar_layout.xml which has the buttons nested in a ViewGroup
+ *
+ * from your Activity call enphasizeButton() method to set state_selected of one of the
+ * buttons.
+ */
+public class ActionBarFragment extends Fragment {
+
+    /**
+     * set the onClickListener for each buttons in the action_bar_layout
+     */
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
+        final View view = inflater.inflate(R.layout.action_bar_layout, container, false);
+
```

Comment:
When you're dealing with fragments it's better to just return the inflated View in the onCreateView method, and then override onViewCreated() where you can initialize all layout elemets. Sometimes you'll need to call certain methods (such as getActivity() to get the context of the activity, which happens here in "mayStartActivity") that are unsafe to call in onCreateView.